### PR TITLE
Meta: Lint code and improve readme

### LINKED ISSUE #4057: Replace `Previously part of Refined GitHub` section in readme with a PR list link
Instead of keeping a list in the readme (which nobody reads because the readme is already hundreds of lines long) we can just a link to: https://github.com/sindresorhus/refined-github/pulls?q=is%3Apr+in%3Atitle+is%3Amerged+-label%3Ameta+drop+feature

Separately, it might be worth going through some old PRs to ensure that they have the "drop feature" in the title.

### PR DESCRIPTION
**Pull request by cheap-glitch**
Closes #4057 

Opening this as a draft in case we notice some more fixes.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -129,6 +129,7 @@ Thanks for contributing! 🦋🙌
 - [](# "repo-wide-file-finder") Enables the File Finder keyboard shortcut (<kbd>t</kbd>) on Issues and Pull Request pages as well.
 - [](# "file-finder-buffer") [Lets you start typing your search immediately after invoking the File Finder (<kbd>t</kbd>), instead of having you wait for it to load first.](https://user-images.githubusercontent.com/1402241/75542106-1c811700-5a5a-11ea-8aa5-bea0472c59e2.gif)
 - [](# "show-associated-branch-prs-on-fork") [Shows the associated pull requests on branches for forked repositories.](https://user-images.githubusercontent.com/16872793/81504659-7e5ec800-92b8-11ea-9ee6-924110e8cca1.png)
+- [](# "html-preview-link") [Adds a link to preview HTML files.](https://user-images.githubusercontent.com/44045911/67634792-48995980-f8fb-11e9-8b6a-7b57d5b12a2f.png)
```
Comment (cheap-glitch) [inline readme.md:132]: Following on https://github.com/sindresorhus/refined-github/pull/4023#discussion_r589109771

**Comment by cheap-glitch** [comment]
Maybe this PR could also take care of #4057 + add guidelines for screenshots as mentioned in https://github.com/sindresorhus/refined-github/pull/4045#discussion_r585558220?

**Comment by yakov116** [comment]
> Maybe this PR could also take care of #4057 

Yes I am almost done

> add guidelines for screenshots as mentioned in [#4045 (comment)](https://github.com/sindresorhus/refined-github/pull/4045#discussion_r585558220)?

👍

**Comment by fregante** [comment]
Can you include this here too? It's just an indentation fix

https://github.com/sindresorhus/refined-github/blob/10d08987fae5fe87cca1a58a1b94807fd4a4d181/source/features/comment-fields-keyboard-shortcuts.tsx#L15-L18

File: readme.md [hunk 2]
```diff
@@ -89,6 +89,15 @@ GitHub Enterprise is also supported. More info in the options.
 - Use smart apostrophes: ’ instead of '
 - Keep it concise.
 
+#####################################
+  Demo screenshots/gifs style guide
+#####################################
+
+- Hide or delete irrelevant DOM elements to give an uncluttered preview of the feature
+- Zoom on the page if the changes are small
+- Re-use the "Before"/"After" arrows from https://user-images.githubusercontent.com/1402241/34438653-f66535a4-ecda-11e7-9406-2e1258050cfa.png
```
Comment (fregante) [inline readme.md:98]:
```suggestion
- If changes are subtle, add a before/after comparison using the arrows from https://user-images.githubusercontent.com/1402241/34438653-f66535a4-ecda-11e7-9406-2e1258050cfa.png
```

File: readme.md [hunk 3]
```diff
@@ -89,6 +89,15 @@ GitHub Enterprise is also supported. More info in the options.
 - Use smart apostrophes: ’ instead of '
 - Keep it concise.
 
+#####################################
+  Demo screenshots/gifs style guide
+#####################################
+
+- Hide or delete irrelevant DOM elements to give an uncluttered preview of the feature
+- Zoom on the page if the changes are small
```
Comment (fregante) [inline readme.md:97]:
```suggestion
- If possible, zoom at 200% before taking a screenshot (native retina screens are fine at 100%)
```

File: readme.md [hunk 4]
```diff
@@ -89,6 +89,15 @@ GitHub Enterprise is also supported. More info in the options.
 - Use smart apostrophes: ’ instead of '
 - Keep it concise.
 
+#####################################
+  Demo screenshots/gifs style guide
+#####################################
+
+- Hide or delete irrelevant DOM elements to give an uncluttered preview of the feature
```
Comment (fregante) [inline readme.md:96]:
```suggestion
- Try cleaning up the page by disabling nearby features or hiding clutter, e.g. everything below this tab bar has been hidden: https://user-images.githubusercontent.com/1402241/55089736-d94f5300-50e8-11e9-9095-329ac74c1e9f.png
```

**Review by fregante — APPROVED** [review]

File: readme.md [hunk 5]
```diff
@@ -347,37 +356,9 @@ Thanks for contributing! 🦋🙌
...
-- [Blog post](https://github.blog/changelog/2019-12-19-improved-attribution-when-squashing-commits/): [Adds `co-authored-by` to the commit when merging PRs with multiple committers.](https://user-images.githubusercontent.com/1402241/51468821-71a42100-1da2-11e9-86aa-fc2a6a29da84.png)
-- [Blog post](https://github.com/sindresorhus/refined-github/pull/3015): [Makes it easier to tell apart commits added to the current PR versus plain commits that reference the PR.](https://user-images.githubusercontent.com/1402241/64478939-398b0a80-d1da-11e9-8c6a-bb98668cb78c.gif)
-- [Blog post](https://github.blog/changelog/2020-06-23-design-updates-to-repositories-and-github-ui/): [Widens the discussion search box.](https://user-images.githubusercontent.com/1402241/55069759-bceaf080-50bf-11e9-84d0-7707de2eb9e9.png)
-
-</details>
+GitHub implemented [dozens of features](https://github.com/sindresorhus/refined-github/pulls?q=is%3Apr+is%3Amerged+label%3A%22implemented+by+github%22) that used to be part of Refined GitHub 🎉
+
+See also [all the dropped features](https://github.com/sindresorhus/refined-github/pulls?q=is%3Apr+is%3Amerged+-label%3Ameta+in%3Atitle+drop+feature).
```
Comment (fregante) [inline readme.md:361]:
```suggestion
[GitHub took inspiration from Refined GitHub](https://github.blog/2018-08-28-announcing-paper-cuts/) and natively implemented dozens of its features, 🎉 so they've been [removed from this extension.](https://github.com/sindresorhus/refined-github/pulls?q=is%3Apr+is%3Amerged+label%3A%22implemented+by+github%22) You can also see [all the past features](https://github.com/sindresorhus/refined-github/pulls?q=is%3Apr+is%3Amerged+-label%3Ameta+in%3Atitle+drop+feature) of Refined GitHub in a single list.
```

**Comment by fregante** [comment]
@yakov116 we already have `title:"dropped feature"`, no need for another label

**Comment by fregante** [comment]
What feature do you think could replace `open-all-conversations` in the highlights table? I don't think it's that :fire: anymore

**Comment by cheap-glitch** [comment]
> What feature do you think could replace `open-all-conversations` in the highlights table?

* Even though I don't use it often, I find `first-published-tag-for-merged-pr` really neat
* Same for `follow-file-renames`
* `hide-markdown-diff` seems relatively popular, given the reactions on [the original PR](https://github.com/sindresorhus/refined-github/pull/3986)

**Comment by yakov116** [comment]
> @yakov116 we already have `title:"dropped feature"`, no need for another label

Sorry I misread 

> I would do two links. "Implemented by GitHub" and "Dropped features".

_Originally posted by https://github.com/sindresorhus/refined-github/issues/4057#issuecomment-792439938_

**Comment by fregante** [comment]
> * Even though I don't use it often, I find [`first-published-tag-for-merged-pr`](/sindresorhus/refined-github/blob/main/source/features/first-published-tag-for-merged-pr.tsx) really neat

Agreed. I rarely see it though because it's so small and doesn't appear in the sticky header.

> * Same for [`follow-file-renames`](/sindresorhus/refined-github/blob/main/source/features/follow-file-renames.tsx)

Indeed. There's even 1 or 2 other extensions for this, but they don't work as well. Maybe we should pick this

> * [`hide-markdown-diff`](/sindresorhus/refined-github/blob/main/source/features/hide-markdown-diff.tsx) seems relatively popular, given the reactions on [the original PR](https://github.com/sindresorhus/refined-github/pull/3986)

You can put a :fire: next to it then

**Comment by yakov116** [comment]
> Agreed. I rarely see it though because it's so small and doesn't appear in the sticky header.

image

In my GH it does 😠

**Review by fregante — APPROVED** [review]
LGTM. Can be merged

**Comment by yakov116** [comment]
@cheap-glitch can you change

https://github.com/sindresorhus/refined-github/blob/86f9ea1fa35a0658202da831dd07c497bbbff605/source/features/sync-pr-commit-title.tsx#L19-L21

to use `getConversationNumber()` please

Thanks

File: source/features/sync-pr-commit-title.tsx [hunk 6]
```diff
@@ -16,13 +17,9 @@ function getCommitTitleField(): HTMLInputElement | undefined {
...
 }
 
-function getPRNumber(): string {
-	return select('.gh-header-number')!.textContent!;
-}
-
 function createCommitTitle(): string {
 	const prTitle = select('.js-issue-title')!.textContent!.trim();
-	return `${prTitle} (${getPRNumber()})`;
+	return `${prTitle} (${getConversationNumber()!})`;
```
Comment (yakov116) [inline source/features/sync-pr-commit-title.tsx:22]:
```suggestion
	return `${prTitle} (#${getConversationNumber()!})`;
```

File: source/features/sync-pr-commit-title.tsx [hunk 7]
```diff
@@ -63,7 +60,7 @@ function updatePRTitle(): void {
 
 	// Remove PR number from commit title
 	const prTitle = getCommitTitleField()!.value
-		.replace(regexJoin(/\s*\(/, getPRNumber(), /\)$/), '');
+		.replace(regexJoin(/\s*\(/, getConversationNumber()!, /\)$/), '');
```
Comment (yakov116) [inline source/features/sync-pr-commit-title.tsx:63]:
```suggestion
		.replace(regexJoin(/\s*\(/, '#' + getConversationNumber()!, /\)$/), '');
```

File: source/options-storage.ts [hunk 8]
```diff
@@ -19,7 +19,6 @@ const defaults = Object.assign({
 
 // TODO[2021-10-01]: Drop classes `muted-link`, `link-gray`, `link-gray-dark`, `text-gray`, `text-gray-light` and `text-gray-dark` #4021
 const migrations = [
-	featureWasRenamed('pr-easy-toggle-files', 'easy-toggle-files'), // Merged in December
```
Comment (fregante) [inline source/options-storage.ts:22]: I think we should start leaving migrations a few more months, they don't really harm anyone. They're useful to who doesn't update super often (due to GHE or whatever reason)

**Pull request merged**